### PR TITLE
lensOn for a particular object

### PR DESCRIPTION
**Pull request by buzzdecafe**
This allows you to bind a lens to a particular object. Should the object come first? or last?

### CODE REVIEW

**Comment by CrossEye** [comment]
What's the use for this?  Especially with `set`, which presumably would usually return a new object?

But if we're going to do it, I would suggest that the object belongs last, and even if we can't use our normal `curry3` function to do it, then the existing `lens` could be considered just curried version of it.

**Comment by buzzdecafe** [comment]
the motivation is mostly to enable composing with lenses using `map`. at present `lens.map` requires you to pass it the object to map over as well as the mapper function. 

i think these are different enough not to just treat them as curried versions of the same thing.

**Comment by CrossEye** [comment]
makes sense.

:herb:

File: dist/ramda.js [hunk 1]
```diff
@@ -3265,6 +3265,41 @@
...
+     *         also curried functions.
+     * @example
+     *
+     *     var xo = {x: 1};
+     *     var xoLens = R.lensOn(xo,
+     *                           function get(o) { return o.x; },
+     *                           function set(v) { return {x: v}; });
+     *     xoLens(); //=> 1
+     *     xoLens.set(1000) //=> {x: 1000}
+     *     xoLens.map(R.add(1)) //=> {x: 2}
```
Comment (davidchambers) [inline dist/ramda.js]: Nit: Missing semicolons on the lines above.

Comment (buzzdecafe) [inline dist/ramda.js]: ok. any thoughts on param order? i'm leaning toward object-last myself now

Comment (davidchambers) [inline dist/ramda.js]: Object last feels right to me.

Comment (CrossEye) [inline dist/ramda.js]: yes, definitely object-last.

Comment (buzzdecafe) [inline dist/ramda.js]: yay we all agree for a change. i will amend and resubmit

**Comment by buzzdecafe** [comment]
params corrected

**Comment by davidchambers** [comment]
Could you rebase and resolve the merge conflicts?

**Comment by buzzdecafe** [comment]
rebased -- take that, @CrossEye! :stuck_out_tongue_winking_eye:

**Comment by davidchambers** [comment]
:deciduous_tree:

**Comment by CrossEye** [comment]
I'm trying to take it!  :wink:  But I seem to have a stubborn system that won't let go.

**Pull request merged**